main.rb

```diff
@@ -0,0 +1,118 @@
+# rubocop:disable Style/CaseEquality
+
+module Enumerable
+  def my_each
+    size.times { |counter| yield(to_a[counter]) } if block_given?
+    block_given? ? self : enum_for(__method__)
+  end
+
+  def my_each_with_index
+    size.times { |counter| yield(to_a[counter], counter) } if block_given?
+    block_given? ? self : enum_for(__method__)
+  end
+
+  def my_select
+    return enum_for(__method__) unless block_given?
+
+    selected = []
+    my_each { |item| selected.push(item) if yield(item) }
+    selected
+  end
+
+  def my_all?(*args, &block)
+    all_true = true
+
+    if (arg = args[0])
+      puts 'Warning, the block is being ignored.' if block
+
+      my_each do |item|
+        all_true &= if arg.is_a? Regexp
+                      (arg === item.to_s)
+                    else
+                      arg === item
+                    end
+      end
+    elsif block_given?
+      my_each { |item| all_true &= yield(item) }
+    else
+      all_true = my_all? { |item| item }
+    end
+
+    all_true
+  end
+
+  def my_any?(*args, &block)
+    result = false
+    if (arg = args[0])
+      puts 'Warning, the block is being ignored. ' if block
+
+      my_each { |item| result |= (arg.is_a?(Regexp) ? (arg === item.to_s) : (arg === item)) }
+    elsif block_given?
+      my_each { |item| result |= yield(item) }
+    else
+      return my_any? { |item| item }
+    end
+    result
+  end
+
+  def my_none?(*args, &block)
+    !my_any?(*args, &block)
+  end
+
+  def my_count(*args, &block)
+    counter = 0
+    if (arg = args[0])
+      puts 'Warning, the block is being ignored. ' if block
+
+      my_each { |item| counter += 1 if item == arg }
+    elsif block_given?
+      my_each { |item| counter += 1 if yield(item) }
+    else
+      my_each { |_item| counter += 1 }
+    end
+    counter
+  end
+
+  def my_map(*args)
+    return enum_for(__method__) if !block_given? && args[0].nil?
+
+    new_array = []
+    my_each { |item| new_array << (args[0].nil? ? yield(item) : args[0].call(item)) }
+    new_array
+  end
+
+  def my_map!(*args)
+    return enum_for(__method__) if !block_given? && args[0].nil?
+
+    my_each_with_index { |item, index| self[index] = (args[0].nil? ? yield(item) : args[0].call(item)) }
+    self
+  end
+
+  def my_inject(*args)
+    if args.size == 2
+      memo = args[0]
+      my_each { |item| memo = memo.send(args[1], item) }
+    elsif args.size == 1 && !block_given?
+      first_el = shift
+      memo = first_el
+      my_each { |item| memo = memo.send(args[0], item) }
+      unshift(first_el)
+
+      return memo
+    else
+      first_el = shift
+      memo = args[0] || first_el
+      my_each { |item| memo = yield(memo, item) if block_given? }
+      unshift(first_el)
+
+      return memo
+    end
+    memo
+  end
+end
```

Comment:
- [ ] should have working `my_inject` identical and returns the same thing as ruby's [`inject`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-inject). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_inject.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

- my_inject should raise a "LocalJumpError" when no block or argument is given
```ruby
[1,5,10,20].my_inject == [1,5,10,20].inject
```
Yours (incorrect)
![image](https://user-images.githubusercontent.com/52209067/83839142-6bbd9000-a6b8-11ea-917d-a56104b01c26.png)

Real inject (correct)
![image](https://user-images.githubusercontent.com/52209067/83839168-7b3cd900-a6b8-11ea-900b-0ce90cad0f4c.png)

- my_inject, when a block is given without an initial value, combines all elements of enum by applying a binary operation, specified by a block::Range
```ruby
(1..20).my_inject { |prod, n| prod * n } == (1..20).inject { |prod, n| prod * n }
```
Yours (incorrect)
![image](https://user-images.githubusercontent.com/52209067/83839874-711bda00-a6ba-11ea-9f1c-e78a03752d6d.png)

Real inject (correct)
![image](https://user-images.githubusercontent.com/52209067/83839919-8264e680-a6ba-11ea-8ed6-5d8226512ff9.png)

- my_inject, when a block is given with an initial value, combines all elements of enum by applying a binary operation, specified by a block
```ruby
(1..20).my_inject(4) { |prod, n| prod * n } == (1..20).inject(4) { |prod, n| prod * n }
```
Yours (incorrect)
![image](https://user-images.githubusercontent.com/52209067/83840113-fef7c500-a6ba-11ea-90d5-55e747c407e3.png)

Real inject (correct)
![image](https://user-images.githubusercontent.com/52209067/83840143-0dde7780-a6bb-11ea-9067-1ddabecca869.png)

- my_inject, when a symbol is specified without an initial value, combines each element of the collection by applying the symbol as a named method:Range
```ruby
(1..20).my_inject(:*) == (1..20).inject(:*)
```
Yours (incorrect)
![image](https://user-images.githubusercontent.com/52209067/83840311-74639580-a6bb-11ea-8a01-234dfd443e86.png)

Real inject (correct)
![image](https://user-images.githubusercontent.com/52209067/83840321-82191b00-a6bb-11ea-8f92-d928ca705e7e.png)

Reply:
Fixed